lizard_ext/lizardio.py

```diff
@@ -1,50 +1,86 @@
 '''
 Fan in and Fan out (again)
 '''
+from collections import Counter
 from .extension_base import ExtensionBase
 
 
+def get_all_indices(value, token_list):
+    indices = []
+    idx = -1
+    while True:
+        try:
+            idx = token_list.index(value, idx + 1)
+            indices.append(idx)
+        except ValueError:
+            break
+    return indices
+
+
 class LizardExtension(ExtensionBase):
     '''
+        Runs the Lizard Extension for the calculation of structural
+        fan-in and fan-out of every procedures in the source code.
+
+        Explanation of terms:
+        'ExtensionBase': Base class for all lizard extensions
+
+        Calculates:
+                'fan_in':                     Number of structural fan-in
+                'fan_out':		              Number of structural fan-out
+                                              in the local scope
+                'general_fan_out'             Number of structural fan-out
+                                              in the global scope
     '''
+
     FUNCTION_INFO = {
-            "fan_in":  {
-                "caption": " fan_in ",
-                "average_caption": " avg_fan_in ",
-                "regression": True},
-            "fan_out": {
-                "caption": " fan_out ",
-                "average_caption": " avg_fan_out ",
-                "regression": True},
-            "general_fan_out": {
-                "caption": " general_fan_out ",
-                "average_caption": " general_fan_out ",
-                "regression": True}
-            }
+        "fan_in": {
+            "caption": " fan_in ",
+            "average_caption": " avg_fan_in ",
+            "regression": True},
+        "fan_out": {
+            "caption": " fan_out ",
+            "average_caption": " avg_fan_out ",
+            "regression": True},
+        "general_fan_out": {
+            "caption": " general_fan_out ",
+            "average_caption": " general_fan_out ",
```

Comment:
@terryyin I made a mistake on the average_caption's name of general_fan_out :) 
Sorry about that.